packages/app/components/Nav/Settings.tsx

```diff
@@ -1,161 +1,64 @@
-  }
-  if (!("PushManager" in window)) {
-    throw new Error("No Push API Support!");
-  }
-};
-
-const requestNotificationPermission = async () => {
-  if (Notification.permission === "granted") {
-    console.log(`permission previously granted`);
-  } else if (Notification.permission === "denied") {
-    console.log("permission previously denied");
-  } else if (Notification.permission === "default") {
-    console.log("permission not set > requesting");
-    await window.Notification.requestPermission();
-  }
-};
-
-export default function Settings() {
-  // first get initial data using useSWR (or from profile)
-  // todo: replace these constants with hits to useSWR call to profile
-  const { data: profile, error, mutate } = useSWR(`api/v1/profile`, async () =>
-    API.profile.index()
-  );
-
-  const [form] = Form.useForm();
-  const [isOpen, setIsOpen] = useState(false);
-
-  const editProfile = async (updateProfile: UpdateProfileParams) => {
-    const newProfile = { ...profile, ...updateProfile };
-    mutate(newProfile, false);
-    await API.profile.patch(
-      pick(newProfile, [
-        "desktopNotifsEnabled",
-        "phoneNotifsEnabled",
-        "phoneNumber",
-      ])
-    );
-    if (updateProfile.desktopNotifsEnabled) {
-      await requestNotificationPermission();
-      // get rid of old service worker, and then try and re-register.
-      // just kidding, this breaks Chrome for some reason (ai ya).
-      // unregister();
-      // have to use setTimeout because unregister does async things, but is sync
-      setTimeout(() => {
-        register();
-      }, 500);
-    }
-    mutate();
-  };
+export default function Settings(): ReactElement {
+  const profile = useProfile();
+  const [isNotifOpen, setIsNotifOpen] = useState(false);
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
 
   return (
     <>
-      <Modal
-        title="Your Settings"
-        visible={isOpen}
-        onCancel={() => setIsOpen(false)}
-        onOk={async () => {
-          const value = await form.validateFields();
-          try {
-            await editProfile(value);
-            setIsOpen(false);
-          } catch (e) {
-            if (
-              e.response?.status === 400 &&
-              e.response?.data?.message === "phone number invalid"
-            ) {
-              form.setFields([
-                { name: "phoneNumber", errors: ["Invalid phone number"] },
-              ]);
-            }
-          }
-        }}
-      >
-        {profile && (
-          <Form form={form} initialValues={profile}>
-            <Form.Item
-              label="Web Notifications"
-              name="desktopNotifsEnabled"
-              valuePropName="checked"
+      <NotificationSettingsModal
+        visible={isNotifOpen}
+        onClose={() => setIsNotifOpen(false)}
+      />
+      <NoPaddingPopover
+        content={
+          <Menu mode="inline">
+            <Menu.Item
```

Comment:
Small UI thing, but the "Notification Settings" menu item stays highlighted after you click on it, and even after closing